### PR TITLE
Add nucleotide tokenizer for llama3 model to recipes

### PR DESCRIPTION
**Pull request by savitha-eng**
### Description

This PR adds a nucleotide tokenizer for the llama3 model, following HuggingFace's PreTrainedTokenizerFast pattern with Rust backend for performance. It follow's NeMo special token conventions (i.e., for EOS, BOS, PAD etc.). 

**Key features:**
- ASCII-based tokenizer that maps nucleotides (A, T, C, G, N, R, Y) to their ASCII values
- NeMo-style special tokens: EOS=0, PAD=1, BOS=2, UNK=3
- Pre-built tokenizer files in `nucleotide_fast_tokenizer/` directory
- `create_tokenizer.py` script for reproducibility
- Comprehensive unit tests covering tokenization, padding, and attention masks

This tokenizer will be used in genomic data loading pipelines for training llama3 models on nucleotide sequences.

#### Usage

```python
from transformers import AutoTokenizer

# Load the nucleotide tokenizer
tokenizer = AutoTokenizer.from_pretrained("bionemo-recipes/models/llama3/nucleotide_fast_tokenizer")

# Tokenize a nucleotide sequence
sequence = "ATCGATCG"
encoded = tokenizer.encode(sequence, add_special_tokens=True)
# Returns: [2, 65, 84, 67, 71, 65, 84, 67, 71, 0]
# [BOS, A, T, C, G, A, T, C, G, EOS]

# Batch tokenization with padding
batch = tokenizer(["AAAA", "TTTTTTTT"], padding=True, return_tensors="pt")
# Returns: {'input_ids': tensor([[2, 65, 65, 65, 65, 0, 1, 1, 1, 1],
#                                  [2, 84, 84, 84, 84, 84, 84, 84, 84, 0]]),
#           'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 0, 0, 0, 0],
#                                      [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])}
```

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Documentation update
- [ ] Other (please describe):

### CI Pipeline Configuration

Configure CI behavior by applying the relevant labels. By default, only basic unit tests are run.

- [ciflow:skip](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/main/contributing/contributing.md#ciflow:skip) - Skip all CI tests for this PR
- [ciflow:notebooks](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/main/contributing/contributing.md#ciflow:notebooks) - Run Jupyter notebooks execution tests for bionemo2
- [ciflow:slow](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/main/contributing/contributing.md#ciflow:slow) - Run slow single GPU integration tests marked as @pytest.mark.slow for bionemo2
- [ciflow:all](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/main/contributing/contributing.md#ciflow:all) - Run all tests (unit tests, slow tests, and notebooks) for bionemo2. This label can be used to enforce running tests for all bionemo2.
- [ciflow:all-recipes](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/main/contributing/contributing.md#ciflow:all-recipes) - Run tests for all recipes (under bionemo-recipes). This label can be used to enforce running tests for all recipes.

Unit tests marked as `@pytest.mark.multi_gpu` or `@pytest.mark.distributed` are not run in the PR pipeline.

For more details, see [CONTRIBUTING](CONTRIBUTING.md)

> [!NOTE]
> By default, only basic unit tests are run. Add appropriate labels to enable an additional test coverage.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

- If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
- If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Pre-submit Checklist

- [x] I have tested these changes locally
- [x] I have updated the documentation accordingly
- [x] I have added/updated tests as needed
- [x] All existing tests pass successfully

### CODE REVIEW

File: bionemo-recipes/models/llama3/create_tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        "<PAD>": pad_id,
+        "<UNK>": unk_id,
+    }
+    
+    if bos_id is not None:
+        special_tokens["<BOS>"] = bos_id
+    
+    # Build vocab: Map each ASCII character to its ord() value
+    # IMPORTANT: Exclude chr(0-3) to reserve those IDs for special tokens
+    vocab = {**special_tokens}
```
Comment (pstjohn) [inline bionemo-recipes/models/llama3/create_tokenizer.py]: just copying the dictionary? do we really need to do that though?

File: bionemo-recipes/models/llama3/create_tokenizer.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    reserved_ids = {eos_id, pad_id, unk_id}
+    if bos_id is not None:
+        reserved_ids.add(bos_id)
```
Comment (pstjohn) [inline bionemo-recipes/models/llama3/create_tokenizer.py]: another nit, but why duplicate this logic? `set(special_tokens.values())` would work, right?

File: bionemo-recipes/models/llama3/create_tokenizer.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    
+    for i in range(256):
+        if i not in reserved_ids:
+            char = chr(i)
+            vocab[char] = i
```
Comment (pstjohn) [inline bionemo-recipes/models/llama3/create_tokenizer.py]:
```python
vocab = {chr(i): i for i in range(256) if i not in reserved_ids}
vocab = {**vocab, **special_tokens}
```

File: bionemo-recipes/models/llama3/create_tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        # Load with AutoTokenizer
+        loaded = AutoTokenizer.from_pretrained(save_path)
+        logger.info("Loaded with AutoTokenizer.from_pretrained()")
+        
+        # Verify it works
+        test_seq = "ATCG"
+        test_enc = loaded.encode(test_seq, add_special_tokens=True)
+        test_dec = loaded.decode(test_enc, skip_special_tokens=True)
+        logger.info(f"Test: '{test_seq}' -> {test_enc} -> '{test_dec}'")
+        logger.info("Loaded tokenizer verified")
```
Comment (pstjohn) [inline bionemo-recipes/models/llama3/create_tokenizer.py]: we should just test this in the test suite right? idk if we need this here

File: bionemo-recipes/models/llama3/create_tokenizer.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    logger.info(f"    UNK: {tokenizer.unk_token} = {tokenizer.unk_token_id}")
+    
+    # Test encoding/decoding
+    logger.info("\n" + "-"*80)
+    logger.info("Test 1: Encoding/Decoding")
+    logger.info("-"*80)
+    
+    sequence = "ATCGATCG"
+    encoded = tokenizer.encode(sequence, add_special_tokens=True)
+    decoded = tokenizer.decode(encoded, skip_special_tokens=True)
```
Comment (pstjohn) [inline bionemo-recipes/models/llama3/create_tokenizer.py]: just way too much logging 😅 . honestly i'd prefer we just pick whether we're using a BOS or not, and make this script as short and simple as possible

**Review by pstjohn — APPROVED** [review]

**Comment by pstjohn** [comment]
/ok to test [bc7100f](https://github.com/NVIDIA/bionemo-framework/pull/1314/commits/bc7100fc68898c160ab9f55e2a0759799a6fdf8d)
